src/report/report-generate.controller.js

```diff
@@ -0,0 +1,286 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+(function() {
+
+    'use strict';
+
+    /**
+     * @ngdoc controller
+     * @name report.controller:ReportGenerateController
+     *
+     * @description
+     * Controller for report options page.
+     */
+    angular
+        .module('report')
+        .controller('ReportGenerateController', controller);
+
+    controller.$inject = [
+        '$state', '$scope', '$window', 'report', 'reportFactory',
+        'reportParamsOptions', 'reportUrlFactory', 'accessTokenFactory',  '$q'
+    ];
+
+    function controller($state, $scope, $window, report, reportFactory,
+                        reportParamsOptions, reportUrlFactory, accessTokenFactory, $q) {
+        var vm = this;
+
+        vm.$onInit = onInit;
+
+        vm.downloadReport = downloadReport;
+
+        vm.paramsInfo = {
+            GeographicZone: 'report.geographicZoneInfo',
+            DueDays: 'report.dueDaysInfo'
+        };
+
+        // SELVSUP-6: Create Stock on Hand Report on reports page
+        vm.booleanOptions = [{
+            name: 'Yes',
```

Comment:
Shouldn't it be a translated value instead of a hardcoded string? Even if this implementation doesn't use other languages it's a good practice and makes it easy to change later. There might even be general translation keys for "Yes" and "No" in the system already.

Reply:
I replaced hardcoded string using message service